Tutorial: How to add session replays to Zendesk (#7733)
PostHog/posthog.com: contents/tutorials/zendesk-session-replays.md

---
title: How to add session replays to Zendesk
date: 2024-02-07
author: ["ian-vanagas"]
tags: ['session replay']
---

Session replays can be a useful support tool for debugging and recreating issues. The errors, console, and network data along with the rest of PostHog's tools make PostHog a powerful support platform.

To get easy access to session replays in Zendesk, you can add a link to them when users submit them. To show you how to do this, we build a basic Next.js app with a form, add PostHog, connect the form to Zendesk, and setup the session replay link in the submitted ticket.

> ## Creating a basic Next.js app with a form

Reviewer: I prefer numbered headings like this. It makes the tutorial easy to follow and easier to find the information you need. I think it also increases the likelihood Google will show you in a featured snippet.

 Feel free to ignore if you prefer not to have it

```suggestion
## 1. Create a basic Next.js app with a form
```

First, make sure [Node is installed](https://nodejs.dev/en/learn/how-to-install-nodejs/) (18.17 or newer), and then create a Next.js app:

```bash
npx create-next-app@latest zendesk-sessions
```

Select **No** for TypeScript, **Yes** for `use app router`, and the defaults for every other option. Once created, go into your `app/page.js` file and set up a basic form for users to submit.

```js
// app/page.js
"use client"
import { useState } from 'react';
import styles from "./page.module.css";

export default function Home() {
  const [content, setContent] = useState('');

  const handleSubmit = async (e) => {
    e.preventDefault();

>     console.log(content)

Reviewer: ```suggestion
```

  }

  return (
    <main className={styles.main}>
      <h1>Submit Your Ticket</h1>
      <form onSubmit={handleSubmit}>
        <textarea 
          value={content} 
          onChange={(e) => setContent(e.target.value)} 
        />
        <br />
        <button type="submit">Submit</button>
      </form>
    </main>
  );
}
```

> Once done, run `npm run dev` and go `http://localhost:3000`  to see your page.

Reviewer: ```suggestion
Once done, run `npm run dev` and go `http://localhost:3000`  to see your app.
```

![Ticket in Zendesk](../images/tutorials/zendesk-session-replays/ticket.png)

## Adding the session replay link to the ticket

The final piece to set up is adding a link to the session replay in PostHog. The JavaScript SDK makes the easy with the `get_session_replay_url()` method. We just add PostHog and slightly modify the content value we pass to `submitTicket` .

```js
"use client"
import { useState } from 'react';
import styles from "./page.module.css";
import { usePostHog } from 'posthog-js/react'

export default function Home() {
  const [content, setContent] = useState('');

  const posthog = usePostHog();

  const handleSubmit = async (e) => {
    e.preventDefault();
    console.log(posthog.get_session_replay_url())
    const finalContent = `
      ${content}\n\n
      Session Replay: ${posthog.get_session_replay_url()}
    `
    await submitTicket(finalContent);
  }
// ... rest of your code 
```

Now, when you check your ticket in Zendesk, you'll see a session replay link to take you to that session in PostHog.

![Zendesk to session replay video](../images/tutorials/zendesk-session-replays/zendesk.mp4)

> > **Bonus:** Beyond a session replay link, you can customize this pattern with information like:

Reviewer: Im not sure what we mean by "this pattern" here
